ui/appui-react/src/appui-react/ui-items-provider/UiItemsManager.ts

```diff
@@ -211,42 +214,57 @@ export class UiItemsManager {
     usage: ToolbarUsage,
     orientation: ToolbarOrientation
   ): ReadonlyArray<ProviderItem<ToolbarItem>> {
-    if (this._abstractAdapter)
-      return this._abstractAdapter.getToolbarButtonItems(
+    const items: ProviderItem<ToolbarItem>[] = [];
+    if (this._abstractAdapter) {
+      const abstractItems = this._abstractAdapter.getToolbarButtonItems(
         stageId,
         stageUsage,
         usage,
         orientation
       );
+      items.push(...abstractItems);
+    } else {
+      UiItemsManager._registeredUiItemsProviders.forEach((entry) => {
+        if (!this.allowItemsFromProvider(entry, stageId, stageUsage)) return;
 
-    const buttonItems: ProviderItem<ToolbarItem>[] = [];
-
-    UiItemsManager._registeredUiItemsProviders.forEach(
-      (entry: UiItemProviderEntry) => {
         const uiProvider = entry.provider;
         const providerId = entry.overrides?.providerId ?? uiProvider.id;
-        // istanbul ignore else
-        if (
-          uiProvider.provideToolbarItems &&
-          this.allowItemsFromProvider(entry, stageId, stageUsage)
-        ) {
-          uiProvider
-            .provideToolbarItems(stageId, stageUsage, usage, orientation)
-            .forEach((spec: ToolbarItem) => {
-              // ignore duplicate ids
-              if (
-                -1 ===
-                buttonItems.findIndex(
-                  (existingItem) => spec.id === existingItem.id
-                )
-              )
-                buttonItems.push({ ...spec, providerId });
-            });
-        }
-      }
-    );
+        const providerItems = uiProvider.provideToolbarItems
+          ? uiProvider
+              .provideToolbarItems(stageId, stageUsage, usage, orientation)
+              .map((item) => ({ ...item, providerId }))
+          : [];
+        items.push(...providerItems);
+      });
+    }
+
+    // Note: items returned by `getToolbarItems` are not returned when using UiItemsManager from @itwin/appui-abstract@^3.0.0
```

Comment:
Should we support new providers from appui-abstract 4.0.0? (when running w/o adapters). With the current implementation calling `AbstractManager.getWidgets()` and `Manager.getWidgets()` will return different results.

Reply:
Is there even a way to run without adapters? UiItemsManager exist in iTwin.js 4.0, so we are always using adapters. (Most of the code is probably never reached as _abstractAdapter always exist, until we reach iTwin.js 5.0)

Even so, I'm not sure I understand the interrogation, at this point, I feel we are only changing the input, but give the same output from the Manager, so we probably have to tweak it so it returns the same thing regardless of was it requested by `get` or `provide` methods, it should still return the same, so the handling will be in the manager itself (So we should have some map that convert the "layouts" properties to the same "widget" we used to have, for example.)

The `layouts` property appearing in the next version would require adding an alpha method to the Manager itself, so "any" layout can request the widgets, and get all of them, and decide where to put them, but I understood that we werent doing that yet. (as such, when I provide a widget via `get`, the result should be the same as if I had provided it with `provide`)

I finally think that if `get` methods exists, we should only call those, and not the `provide` ones, as we expect whoever to have opted in the `get` methods to want to test them, and the way it is implemented right now, we will get both, and drop one of them (so, it will be difficult to test if it actually works)

Reply:
That's correct - we only use the adapters at the moment.
These are valid points, I'll work on it.

Reply:
Applied the following changes which simplified code quite a bit:
- Use the adapter to map items returned by `get` properties to `provide` properties
- Prefer `get` properties instead of `provide` in `UiItemsProvider`.
- Removed arguments from `getBackstageItems`.
- Removed `stageIds` and `stageUsages`. These were optional either way, we can add them in the future if needed.
- Enhanced the tests to actually test both variants: w/ and w/o abstract adapters

Reply:
Tests added.